trustmanager/keydbstore.go

```diff
@@ -0,0 +1,153 @@
+package trustmanager
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+
+	"github.com/docker/notary/pkg/passphrase"
+	"github.com/endophage/gotuf/data"
+	"github.com/jinzhu/gorm"
+	gojose "github.com/square/go-jose"
+)
+
+// KeyDBStore persists and manages private keys on a SQL database
+type KeyDBStore struct {
+	sync.Mutex
+	db         gorm.DB
+	passphrase string
+	encrypter  gojose.Encrypter
+	cachedKeys map[string]data.PrivateKey
+}
+
+// GormPrivateKey represents a PrivateKey in the database
+type GormPrivateKey struct {
+	gorm.Model
+	KeyID      string `sql:"not null;unique"`
+	Encryption string `sql:"not null"`
+	Algorithm  string `sql:"not null"`
+	Public     []byte `sql:"not null"`
+	Private    string `sql:"not null"`
+}
+
+// TableName sets a specific table name for our GormPrivateKey
+func (g GormPrivateKey) TableName() string {
+	return "private_keys"
+}
+
+// NewKeyDBStore returns a new KeyDBStore backed by a SQL database
+func NewKeyDBStore(passphraseRetriever passphrase.Retriever, dbType string, dbSQL *sql.DB) (*KeyDBStore, error) {
+	cachedKeys := make(map[string]data.PrivateKey)
+
+	// Retreive the passphrase that will be used to encrypt the keys
+	passphrase, _, err := passphraseRetriever("", "", false, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// Setup our encrypted object
+	encrypter, err := gojose.NewEncrypter(gojose.A256GCMKW, gojose.A256GCM, []byte(passphrase))
```

Comment:
Got it. In that case, does it make sense to just ditch the passphrase retriever for this keystore?

If you really wanted to do PBES, we could use this lib instead of go-jose: https://github.com/dvsekhvalnov/jose2go